src/kafkerl_protocol.erl

```diff
@@ -150,17 +150,10 @@ parse_fetch_response(Bin, {Remainder, CorrelationId, Steps}) ->
 parse_metadata_response(<<CorrelationId:?UINT,
                           BrokerCount:?UINT,
                           BrokersBin/binary>>) ->
-  case parse_brokers(BrokerCount, BrokersBin) of
-    {ok, Brokers, <<TopicCount:?UINT, TopicsBin/binary>>} ->
-      case parse_topic_metadata(TopicCount, TopicsBin) of
-        {ok, Metadata} ->
-          {ok, CorrelationId, {Brokers, Metadata}};
-        {error, _Reason} = Error ->
-          Error
-      end;
-    {error, _Reason} = Error ->
-      Error
-  end;
+  {ok, Brokers, <<TopicCount:?UINT, TopicsBin/binary>>} =
```

Comment:
`parse_brokers` never returns any other thing than `{ok, _, binary()}`.
If there is an error to report, that function is not reporting it.